app/src/main/admin/components/AdminForgottenPasswordPage.tsx

```diff
@@ -1,28 +1,34 @@
 import * as React from "react";
-import { AdminPageWithHeader } from "./AdminPageWithHeader";
-import { ForgottenPasswordForm } from "../../shared/components/Login/ForgottenPasswordForm";
-import {IPageWithParent} from "../../shared/models/Breadcrumb";
-import {MainMenu} from "./MainMenu/MainMenu";
-import { Page } from "../../shared/components/PageWithHeader/Page";
+import {ForgottenPasswordForm} from "../../shared/components/Login/ForgottenPasswordForm";
+import {PageBreadcrumb, PageProperties} from "../../shared/components/PageWithHeader/PageWithHeader";
+import {AdminPageHeader} from "./AdminPageHeader";
+import {PageArticle} from "../../shared/components/PageWithHeader/PageArticle";
+import {BreadcrumbInitializer} from "../../shared/components/Breadcrumbs/BreadcrumbsInitializer";
+import {compose} from "recompose";
 
 const pageTitle = "Forgotten your password?";
 
-export class AdminForgottenPasswordPage extends AdminPageWithHeader<undefined> {
-    name(): string {
-        return pageTitle;
+class AdminForgottenPasswordPageComponent extends React.Component<PageProperties<undefined>> {
+    componentDidMount() {
+        this.props.createBreadcrumbs(AdminForgottenPasswordPageComponent.breadcrumb());
     }
 
-    urlFragment(): string {
-        return "forgotten-password/";
-    }
-
-    parent(): IPageWithParent {
-        return new MainMenu();
+    static breadcrumb() : PageBreadcrumb {
+        return {
+            name: pageTitle,
+            urlFragment: "forgotten-password/"
+        }
     }
 
     render(): JSX.Element {
-        return <Page page={this}>
-            <ForgottenPasswordForm />
-        </Page>;
+        return <div>
+            <AdminPageHeader/>
```

Comment:
Am I right in thinking that now everything is redux we can move this `AdminpageHeader` component to the router (similarly for the other portals) and then we could have a single `ForgottenPasswordPageComponent` that all 3 portals route to? I think the only difference between these 3 pages is the header...

Reply:
Feel like this ticket has enough PRs. So here's a new ticket for either of us to work on: https://vimc.myjetbrains.com/youtrack/issue/VIMC-2020